src/middlewared/middlewared/plugins/mail_/gmail.py

```diff
@@ -0,0 +1,51 @@
+import base64
+
+from googleapiclient.discovery import build
+from google.oauth2.credentials import Credentials
+
+from middlewared.service import private, Service
+
+
+class GmailService:
+    def __init__(self, config):
+        self.config = config
+
+        credentials = Credentials.from_authorized_user_info(config["oauth"])
+        self.service = build("gmail", "v1", credentials=credentials)
```

Comment:
Should we also be closing this ? i.e `self.service.close()` when we don't require the `GmailService` object anymore to avoid leaving sockets open ?

Reply:
There is no persistent connection, this resource only handles refreshing OAuth access token periodically (only on request so there is no background thread as well)

Reply:
Thank you for clarifying, i was just having a peek at https://github.com/googleapis/google-api-python-client/blob/master/docs/start.md when i noticed the sockets bit.

Reply:
So we don't need to close it ?

Reply:
@sonicaj good catch actually, thank you